OPERATOR.rst

```diff
@@ -93,14 +94,36 @@ Triaging ``sandbox`` failures
   discretion on whether or not to open a new ticket, it is important to record
   unusual failures to determine their eventual frequency.
 
+Triaging GitLab build failures on ``dev`` and ``prod``
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+If a GitLab build fails on a stable deployment, the operator must evaluate the
+impact of that failure. This evaluation should include visiting the Data Browser
+to verify it isn't broken.
+
+To restore the deployment to a known working state, the operator should rerun
+the deploy job of previous passing pipeline for that deployment. This can be
+done without pushing anything and only takes a couple of minutes. The branch
+for that deployment must then be reverted to the previously passing commit.
+
+
 .. _committing the changes separately: https://github.com/DataBiosphere/azul/issues/2899#issuecomment-804508017
 
 Reindexing
 ^^^^^^^^^^
 
 The operator must check the status of the queues after every reindex for
 failures. Use ``python scripts/manage_queues.py`` to identify any failed
-messages. If failed messages are found
+messages. If failed messages are found, use ``python scripts/manage_queues.py``
+to
+
+- dump the failed messages to JSON file(s)
+- clear the fail queues so they are empty for the next reindex
+- feed the failed messages back into their respective ``_retry`` queues
+
+This may cause the previously failed messages to succeed. Repeat this procedure
+until no progress towards success is being made, i.e., the fail queues are empty
+or no previosuly failed messages succeeded. Finally
```

Comment:
Should state the invariant post condition that all queues must be empty and that any permanent failures are tracked in tickets.